.github/workflows/haskell-ci.yml

```diff
@@ -1,43 +1,44 @@
 name: CI
 
-on: 
+on:
   pull_request:
   push:
     branches: [master]
 
-
 jobs:
   stack:
-    name: stack / resolver ${{ matrix.resolver }} 
+    name: ghc ${{ matrix.ghc }}
     runs-on: ubuntu-latest
     strategy:
       matrix:
         stack: ["2.7.3"]
-        resolver: ["lts-16.2"]
-        ghc: ["8.8.3"]
+        ghc:
+          - 8.6.5
+          - 8.8.4
+          - 8.10.7
 
     steps:
-    - uses: actions/checkout@v2
-      if: github.event.action == 'opened' || github.event.action == 'synchronize' || github.event.ref == 'refs/heads/master'
+      - uses: actions/checkout@v2
+        if: github.event.action == 'opened' || github.event.action == 'synchronize' || github.event.ref == 'refs/heads/master'
 
-    - uses: haskell/actions/setup@v1
-      name: Setup Haskell Stack
-      with:
-        ghc-version: ${{ matrix.ghc }}
-        stack-version: ${{ matrix.stack }}
+      - uses: haskell/actions/setup@v1
+        name: Setup Haskell Stack
+        with:
+          ghc-version: ${{ matrix.ghc }}
+          stack-version: ${{ matrix.stack }}
 
-    - uses: actions/cache@v2.1.6
-      name: Cache ~/.stack
-      with:
-        path: ~/.stack
-        key: ${{ runner.os }}-${{ matrix.resolver }}-stack
+      - uses: actions/cache@v2.1.6
+        name: Cache ~/.stack
+        with:
+          path: ~/.stack
+          key: ${{ runner.os }}-${{ matrix.resolver }}-stack
 
-    - name: Install dependencies
-      run: |
-        stack build --test --bench --no-run-tests --no-run-benchmarks --only-dependencies
-    - name: Build
-      run: |
-        stack build --test --bench --no-run-tests --no-run-benchmarks
-    - name: Test
-      run: |
-        stack test
\ No newline at end of file
+      - name: Install dependencies
+        run: |
+          stack build --test --bench --no-run-tests --no-run-benchmarks --only-dependencies
```

Comment:
If you want these stack commands to use the stack yaml files, you'll have to specify it on cli,
something like `stack build .... --stack-yaml stack-${{ matrix.ghc }}`

Also if you want to use the ghc installed by `haskell/actions/setup@v1` 
you should add `--system-ghc` to all the stack commands (otherwise stack will install its own GHC and ignore the ghc installation provided by that setup action.

And

Reply:
You can see it's downloading ghc-8.8.3 in 8.10.7 job
![Screenshot from 2022-01-04 13-04-39](https://user-images.githubusercontent.com/2716069/148056542-3f8f48be-a501-4f56-9ea4-71be0a538974.png)

Reply:
Thanks for pointing this out. I did not realise this was happening.

Reply:
I like to use lazy approach to this. Rather than being forced to check everything all the time use the workflow where it's hard to miss the problem. If you want to make something build with new version of compiler, start by replicating compilation error in ci. Only then attempt to fix it.